File: mater-webapp/src/app/store/dtos/car.dto.ts
import {AbstractDto} from "./core/abstract.dto";
import {FileDto} from "./file.dto";
import {GearboxEnum} from "../enums/gearbox.enum";
import {CarStateEnum} from "../enums/car-state.enum";
import {EngineOilTypeEnum} from "../enums/engine-oil-type.enum";
import {TransmissionEnum} from "../enums/transmission.enum";
import {SteeringWheelEnum} from "../enums/steering-wheel.enum";

export class CarDto extends AbstractDto {
  brand: string;
  model: string;
  productionYear: number;
  price: number;
  mileage: number;
  VIN: string;
  ownersAmount: number;
  power: number;
  files: FileDto[];
  documentsColor: string;
  gearbox: GearboxEnum;
  state: CarStateEnum;
  engineOilType: EngineOilTypeEnum;
  transmission: TransmissionEnum;
  steeringWheel: SteeringWheelEnum;

  get title(): string {
    return (!!this.brand && !!this.model && !!this.productionYear)
      ? 'Информация о машине загружается...' : `${this.brand} ${this.model} (${this.productionYear})`;
  }
}

File: mater-rest/src/main/java/com/oneune/mater/rest/main/store/dtos/CarDto.java
package com.oneune.mater.rest.main.store.dtos;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.oneune.mater.rest.main.store.dtos.core.AbstractDto;
import com.oneune.mater.rest.main.store.dtos.files.FileDto;
import com.oneune.mater.rest.main.store.enums.*;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;
import lombok.experimental.SuperBuilder;

import java.math.BigDecimal;
import java.util.List;

/**
 * @see com.oneune.mater.rest.main.store.entities.CarEntity
 */
@FieldDefaults(level = AccessLevel.PRIVATE)
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
@Data
public class CarDto extends AbstractDto {
    String brand;
    String model;
    Integer productionYear;
    BigDecimal price;
    BigDecimal mileage;
    @JsonProperty("VIN")
    String VIN;
    Integer ownersAmount;
    Integer power;
    List<FileDto> files;
    String documentsColor;
    GearboxEnum gearbox;
    CarStateEnum state;
    EngineOilTypeEnum engineOilType;
    TransmissionEnum transmission;
    SteeringWheelEnum steeringWheel;
}
